### PR TITLE
Only display featured image UI when theme supports it too

### LINKED ISSUE #2523: Only display Featured Image field if theme supports post thumbnails
A Featured Image field should only be available both if the current post type has `thumbnail` supports (available from [post types endpoints](https://developer.wordpress.org/rest-api/reference/post-types/)) **and the [theme supports post thumbnails](https://codex.wordpress.org/Post_Thumbnails)**. It is not currently possible to retrieve theme supports from the REST API. Alternatives exist, notably bootstrapping this data server-side on page load, but see #2499 for context on why this is not advisable.

### PR DESCRIPTION
**Pull request by danielbachhuber**
## Description

Ensures the "Featured Image" UI only displays when the theme supports `post-thumbnails` too.

Fixes #2523
Previously #6510 
## Types of changes

My code introduces a new `ThemeSupportCheck` component to inspect the theme's `post-thumbnails` setting. If the theme doesn't support post thumbnails, then the featured image UI doesn't render.

The code also introduces `post-thumbnails=true` on the site index for exposure through the REST API.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 

## How has this been tested?

Using Twenty Seventeen, here's the UI that normally displays:

image

Then, add the following code snippet to a `wp-content/mu-plugins/local.php` file:

```
add_action( 'after_setup_theme', function(){
	remove_theme_support( 'post-thumbnails' );
}, 100 );
```

The "Featured Image" will no longer display:

image

### CODE REVIEW

**Comment by danielbachhuber** [comment]
Worth noting: I wanted to put `theme-support-check` in `@wordpress/components`, but doing so apparently causes a dependency issue with `withSelect()`.

**Comment by gziolo** [comment]
That’s neat idea 👍

File: editor/components/post-featured-image/check.js [hunk 1]
```diff
@@ -2,9 +2,12 @@
  * Internal dependencies
  */
 import PostTypeSupportCheck from '../post-type-support-check';
+import ThemeSupportCheck from '../theme-support-check';
 
 function PostFeaturedImageCheck( props ) {
-	return <PostTypeSupportCheck { ...props } supportKeys="thumbnail" />;
+	return <ThemeSupportCheck supportKeys="post-thumbnails">
```
Comment (aduth) [inline editor/components/post-featured-image/check.js]: Nit (Style): My preference (and predominantly throughout the codebase) is for JSX to be placed on its own lines if multi-lined with parentheses, e.g.:

```jsx
return (
	<ThemeSupportCheck supportKeys="post-thumbnails">
		<PostTypeSupportCheck { ...props } supportKeys="thumbnail" />
	</ThemeSupportCheck>
);
```

If it's a convention we want to adopt, we could enforce with:

https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-wrap-multilines.md

**Review by aduth — APPROVED** [review]
Works well 👍

File: editor/components/post-featured-image/check.js (hunk 1 above)
Comment (danielbachhuber) [reply editor/components/post-featured-image/check.js]: > My preference (and predominantly throughout the codebase) is for JSX to be placed on its own lines with parentheses if multi-lined, e.g.:

Seems reasonable to me 19160260e

**Pull request merged**